Sources/_RegexParser/Regex/Parse/LexicalAnalysis.swift

```diff
@@ -578,9 +578,21 @@ extension Source {
   mutating func lexQuote(context: ParsingContext) throws -> AST.Quote? {
     let str = try recordLoc { src -> String? in
       if src.tryEat(sequence: #"\Q"#) {
-        return try src.expectQuoted(endingWith: #"\E"#).value
+        let contents = src.lexUntil { src in
+          src.isEmpty || src.starts(with: #"\E"#)
+        }.value
+
+        // If we have an ending, the sequence shouldn't be empty. This isn't a
+        // particulary useful thing to express, and [\Q\E] should definitely be
+        // illegal. The unbounded case may however be empty.
+        let ending = src.tryEat(sequence: #"\E"#)
+        if ending && contents.isEmpty {
+          throw ParseError.expectedNonEmptyContents
+        }
```

Comment:
@milseman ICU does allow empty `\Q\E` outside of a character class, do you think it's worth supporting that? PCRE doesn't support it, and IMO it doesn't seem particularly useful, but we could support it while rejecting empty `\Q\E` within a character class. Any thoughts?

Reply:
Weakly in favor of allowing it unless you can think of any downsides?

Reply:
Seems reasonable to me, I can't think of any downsides. I've updated the PR to allow them, and also added the rule to reject quotes that span multiple lines.